docarray/array/mixins/evaluation.py

```diff
@@ -161,7 +166,16 @@ def evaluate(
         results = defaultdict(list)
         caller_max_rel = kwargs.pop('max_rel', None)
         for d, gd in zip(self, ground_truth):
-            max_rel = caller_max_rel or len(gd.matches)
+            if caller_max_rel:
```

Comment:
I think you need to refactor the if else logic here a bit
```
for d, gd in zip(self, ground_truth):
    if caller_max_rel:
        max_rel = caller_max_rel

    if ground_truth_type == 'labels':
        if max_rel_per_label:
            max_rel = max_rel_per_label.get(d.tags[label_tag], None)
            if max_rel is None:
                raise ValueError(
                    '`max_rel_per_label` misses the label ' + str(d.tags[label_tag])
                )
        else:
            raise ValueError('max_rel is required or something')
    else:
        max_rel = len(gd.matches)
```

Reply:
I think it is correct, that `caller_max_rel` is used if the user provides a `max_rel` attribute explicitly.

Reply:
This exception when `max_rel` is not set should also not be their because most of the metrics do not require `max_rel`, but setting it to None might be better than setting it to `len(gd.matches)`. I will change this.